fotoapparat/src/main/java/io/fotoapparat/parameter/range/Range.java

```diff
@@ -0,0 +1,36 @@
+package io.fotoapparat.parameter.range;
+
+/**
+ * Created by ychen on 5/25/2017.
+ */
+
+public abstract class Range<T> {
```

Comment:
Prefer `interface` whenever possible (and it is possible here). That simplifies readability when you go through implementations.